### PR TITLE
Reading pull attachments should depend on read UnitTypePullRequests

### LINKED ISSUE #10336: download of attachments is broken if internal issue management is disabled
- Gitea version (or commit ref): 1.11.x **as docker container**
- Git version: the version contained in https://hub.docker.com/r/gitea/gitea
- Operating system: Docker host is Debian 8
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL) : https://try.gitea.io/drahkrub/toast/pulls/1
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

If internal issue management is disabled or an external issue management is used, than download of attachments (at least of attachments attached to pull requests) of the form "[ROOT_URL]/attachments/[:uuid]" produce a 404.

Can not be reproduced in 1.10.4, only in 1.11.0 and 1.11.1.

### PR DESCRIPTION
**Pull request by zeripath**
Previously these would depend on read UnitTypeIssues.

Fixes #10336

### CODE REVIEW

**Review by jolheiser — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

File: models/attachment.go [hunk 1]
```diff
@@ -79,7 +79,11 @@ func (a *Attachment) LinkedRepository() (*Repository, UnitType, error) {
 			return nil, UnitTypeIssues, err
```
Comment (drahkrub) [inline models/attachment.go:79]: Maybe the correct unitType (UnitTypePullRequests instead of UnitTypeIssues) is of interest here, too?!

Comment (zeripath) [reply models/attachment.go:79]: don't have an issue to be able to check if it is a pull

Comment (zeripath) [reply models/attachment.go:79]: and in any case if err is not nil then the unitType is not used: 

```go
repository, unitType, err := attach.LinkedRepository()
	if err != nil {
		ctx.ServerError("LinkedRepository", err)
		return
	}
```

Comment (drahkrub) [reply models/attachment.go:79]: Oh, of course, you're right.

**Pull request merged**